compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1073,12 +1073,6 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                         // for instance
                                         self.tcx.at(span).type_of(*def_id).instantiate_identity()
                                             != rcvr_ty
-                                            && self
-                                                .tcx
-                                                .at(span)
-                                                .type_of(*def_id)
-                                                .instantiate_identity()
-                                                != rcvr_ty
```

Comment:
@compiler-errors you updated this https://github.com/rust-lang/rust/commit/3f697b85f2196ea7d2a5d4fa1f88c39e63871d39, but I'm not sure changing the use of `actual` to `rcvr_ty` is correct here. My read of the comment above suggests we explicitly wanted to compare the type before and after `resolve_vars_if_possible`. 

Am I misreading this?

Reply:
I don't believe it makes sense to compare before `rcvr_ty` has been resolved, so it makes sense to remove this branch. At least according to that commit, it was not actually ever being exercised either?

Reply:
Looks like this has been essentially unchanged since it was added in #84221 which didn't add a test that covers the `Box<Self>` case the comment talks about. 

I don't have an objection to removing the check, just thought I'd flag it as a bit suspicious 🙂